### ISSUE TITLE
Unexpected behaviours on the spatial tab

### ISSUE THREAD

**Issue by mialondon** [scholarslab/Neatline#178]
I've had trouble replicating it when logged in as Super, but I've had more luck when logged in as a regular (i.e. Admin) users.

The issues I've been able to replicate are:
1. Create a Neatline item; add title; save.  All fine.
2. Go to the Spatial tab.  Draw a polygon or drop a marker.  Save.  All fine.
3. Try to edit the item's spatial attributes by adding a new marker or polygon - this is where it gets odd.  Phantom markers appear, possibly based on mouse movements, though not based on mouse clicks (e.g. the mouse was just moving around over the map, I wasn't clicking, holding or dragging anywhere). On the screenshot, the 'phantoms' are orange, the intentionally added coordinates are black https://www.dropbox.com/s/t76d1m548vo9prc/phantom%20markers.png  They don't show up on the public version http://www.digitalhistories.org.uk/neatline-exhibits/show/clean-test-version

Another issue I noticed was that when I click into a different Neatline item, the map still shows the view and markers for the earlier item e.g. https://www.dropbox.com/s/gebnonrfn3ogcl3/Different_item_showing_map_and_markers_for_earlier_item.png though as you can see the spatial box itself is empty.

I haven't been able to replicate all the bigger issues I saw others have, where the main issue seems to be that nothing they do with the coordinate controls seems to be saved, but I figured this was a start.

In general I've noticed that changing the base layer has odd side-effects - e.g. the zoom controls disappear under the items bar.  I don't think I've played with the base layers on this test exhibit, but I demonstrated it during the workshop so that's another potential factor.

I've noticed that the 'delete' button only rarely works, but I'm not sure if I'm meant to something get browser focus on the coordinate I want to delete.  I also don't seem to be able to delete items in the Text tab either, but again, perhaps I'm misunderstanding what it's meant to delete.

I'm currently on Firefox 16.0.2, running on a laptop with Windows 7 Pro, SP 1.  digitalhistories.org.uk is hosted on Dreamhost, with Omeka installed with the one-click installer then upgraded.

Cheers, Mia

**Comment by mialondon** [comment]
Ok, changing the basemap didn't even trigger the errors it's triggered other times, though I'm logged in on the lower level account.

**Comment by davidmcclure** [comment]
Hi Mia,
- **Phantom dots in the editor**: Yep, those are artifacts of the OpenLayers editing controls - when the user activates one of the editing tools (point, poly, line), draws something, and then moves the cursor off of the map (eg, to save the record), the little dot that trails the cursor gets "stuck" at the edge of the map. I remember seeing this in development, and thought we had fixed it - shouldn't be hard to iron out. 
  
  This is actually just a cosmetic problem endemic to the editor - the dots aren't getting added to the saved spatial data (they won't appear in the real exhibit) and they will disappear if you refresh the page in the editor.
- **Spatial geometry for Record A visible when Record B is being edited**: If I'm understanding right, I think what you're describing is deliberate. When you click on one of the record listings in the left pane, that "activates" that record for editing, meaning that you can add/edit/delete map geometries that correspond to that vector. Geometries for other records remain visible - this is important in complex exhibits where you need to see exactly where/how other records are annotated as you draw annotations for another record.
- **Layer switcher getting occluded by items**: This is a know problem (see #176). We're actually making a lot of changes in the upcoming release that simplify the whole layout process, so this won't be a problem after that.
- **Delete button**: Do you mean the record delete button (at the bottom of a record edit form, next to "Save") or the geometry delete button (on the map, top left, next to rotate, drag, etc.)? If the map delete - that button deletes the _currently selected_ vector on the map. So, if you have a record with 5 geometries, you could delete 1 of them by clicking on the geometry you want to get rid of (it will change color, and the points will turn into drag handles) and clicking the delete button, which will remove the geometry. Save the record as usual to commit the change.

**Comment by mialondon** [comment]
Hi David,

re phantom dots - I saw that a lot in the workshop but hadn't seen it on either of my two installations on different machines (the laptop above and a Mac running OSX 10.5.8, usually with the latest Firefox).  Good to know there's a fix!

I realised afterwards that the spatial geometry of two records is probably deliberate - I think it's mostly just confusing because of all the other issues, but I do also wonder if people will realise they are switching items when they click on different geometries?

Are you meant to be able to select a geometry once you've clicked off it?  I very occasionally seemed to be able to so I could delete or move things around, but not reliably so.  Does it behave differently when you've newly selected an item?

The biggest issue that people reported is still the non-appearing geometries.  It might be worth asking Anouk what browser/OS combination she uses, and what was used on the computers in the lab where the workshop was held in case that's a factor.

Cheers, Mia
